dll/shellext/fontext/CEnumFonts.cpp

```diff
@@ -48,15 +49,25 @@ class CEnumFonts :
             if (m_Index < g_FontCache->Size())
             {
                 CStringW Name = g_FontCache->Name(m_Index);
-                rgelt[Fetched] = _ILCreate(Name, m_Index);
-
+                LPITEMIDLIST item = _ILCreate(Name, m_Index);
+                if (!item)
+                {
+                    hr = Fetched ? S_FALSE : E_OUTOFMEMORY;
```

Comment:
How could this be `S_FALSE?

Reply:
> How could this be `S_FALSE`?

If the caller asks for more than one item and you successfully fetch the first item, then `rgelt[0]` contains a valid item and the return value can only be S_OK or S_FALSE.